app/components/EntityForm/index.js

```diff
@@ -0,0 +1,45 @@
+import React, { PropTypes } from 'react';
+import { Control, Form } from 'react-redux-form/immutable';
+
+class EntityForm extends React.Component { // eslint-disable-line react/prefer-stateless-function
+  render() {
+    return (
+      <Form
+        model={this.props.model}
+        onSubmit={this.props.handleSubmit}
+      >
+        {
+          this.props.fields.map((el, index) => (
+            <span key={index}>
+              <label htmlFor={el.id}>{el.label}</label>
+              { el.type === 'text' &&
+                <Control.text id={el.id} model={el.model} />
+              }
+              { el.type === 'textarea' &&
+                <Control.textarea id={el.id} model={el.model} />
+              }
+              { el.type === 'select' &&
+                <Control.select id={el.id} model={el.model}>
+                  {
+                    el.options.map((option, i) =>
+                      <option key={i} value={option.value}>{option.label}</option>
+                    )
+                  }
+                </Control.select>
+              }
```

Comment:
I thought of having a function here with a way to define a custom component that you want rendered, as I find its always good to have that escape hatch.

But I think I might prefer the way you have this setup as its easier to read...
```
const components = {
  input: Control.input,
  textarea: Control.textarea,
  radio: Control.radio,
  checkbox: Control.checkbox,
  file: Control.file,
  select: Control.select,
  button: Control.button,
};


getFieldComponent = (field) => {
    if (field.component) {
      return field.component;
    } else if (field.controlType in components) {
      return components[field.controlType];
    }
    return components.input;
  } 

 const FieldComponent = this.getFieldComponent(field);

```